### DISCUSSION TITLE
How do you make a retro style blur?

### DISCUSSION THREAD

**Discussion by Bluebuster116** [ImageMagick/ImageMagick#5129] [Consulting]
Is IM capable of doing this?. I am not sure if you know my meaning. I mean back in the day when they wanted to blur something on tv they made a certain area pixelated. There is a filter for this in IM but it doesn't produce this result.

It might be helpful if somebody could tell me how to do it with some example of the result. Please attach some picture with syntax.

**Comment by fmw42** [comment]
I have a script that would pixelate. But the concept is very simply, just -scale down and then -scale back up again. So you can make a mask of the area you want. Then pixelate the whole image. Then use the mask to mix between the two images so it uses the pixelated version for where the mask is white and original image where the mask is black.

**Comment by Bluebuster116** [comment]
@fmw42

I don't have the sufficient experience on IM to do that on my own. Can you show me an example please?.

It appears to be that there isn't a straight way to do that but rather in a sequence of steps. Perhaps does it exist some work around to do it in one line of code?

Otherwise please teach me how would you do it? and if possible a picture of how the result looks like?.

**Comment by fmw42** [comment]
```
Read image
Make a copy, then scale down by 1/8=12.5% (or whatever reduction you want), then scale up to original size
Read mask
Composite the original image and pixelated image using the mask
Save the result

```
**On Unix:**

Imagemagick 6 code:

Input:

![lena](https://user-images.githubusercontent.com/13458218/167270730-b7e15d0a-9939-4e1b-9a7a-811008a2d30f.png)

Mask:

![mask](https://user-images.githubusercontent.com/13458218/167270803-8e26081b-56fd-4195-b625-491658f9315b.png)



```
convert lena.png \
\( +clone -scale 12.5%x12.5% -scale 256x256\! \) \
mask.png \
-compose over -composite \
lena_pixelate.png

```

![lena_pixelate](https://user-images.githubusercontent.com/13458218/167270734-5fa6a8a5-950b-45c3-8174-3c2855abc5eb.png)

If using Imagemagick 7, then change convert to magick.


See 

https://legacy.imagemagick.org/Usage/resize/#scale
https://legacy.imagemagick.org/Usage/basics/#parenthesis
https://legacy.imagemagick.org/Usage/compose/#compose

If you want it blurry, then use -resize in place of the second -scale

```
convert lena.png \
\( +clone -scale 12.5%x12.5% -resize 256x256\! \) \
mask.png \
-compose over -composite \
lena_pixelate2.png

```

![lena_pixelate2](https://user-images.githubusercontent.com/13458218/167271036-16179841-1d0b-4169-8c6a-4a5d58255034.png)


**Windows Syntax is different.**

```
convert lena.png ^
( +clone -scale 12.5%x12.5% -resize 256x256^! ) ^
mask.png ^
-compose over -composite ^
lena_pixelate2.png

```

**Comment by fmw42** [comment]
For novices, see

http://www.imagemagick.org/discourse-server/viewtopic.php?f=1&t=9620
http://www.imagemagick.org/script/command-line-processing.php
https://imagemagick.org/script/command-line-options.php
http://www.imagemagick.org/Usage/reference.html
http://www.imagemagick.org/Usage/
https://github.com/ImageMagick/usage-markdownhttps://imagemagick.org/script/porting.php#cli
https://legacy.imagemagick.org/discourse-server/index.php
